apps/common/lib/lexical/source_file/store.ex

```diff
@@ -145,6 +146,48 @@ defmodule Lexical.SourceFile.Store do
 
     defp normalize_error(:error), do: {:error, :not_open}
     defp normalize_error(e), do: e
+
+    @read_types [:sources, :temp, :any]
+    @write_types [:sources, :temp]
+    defp ets_fetch(type, key) when type in @read_types do
+      case :ets.match(@table_name, {key, type_selector(type), :"$1"}) do
+        [[value]] -> {:ok, value}
+        _ -> :error
+      end
+    end
+
+    defp ets_put(type, key, value) when type in @write_types do
+      :ets.insert(@table_name, {key, type, value})
```

Comment:
All internal function accept `type, key` as the order of arguments, yet they're stored in `{key , type ..}` in ETS. Perhaps it would be better to use the same order? The info in ETS could also be used by other, for example plugins?

Reply:
I did that because of a micro optimization with pattern matching on the first argument. I typically like to make the first arg the one matched against. It’s likely not necessary here